tests/RecordTests/FunctionalTest/k4a_cpp_ft.cpp

```diff
@@ -0,0 +1,453 @@
+        ASSERT_GE(version.rgb.major, (uint32_t)1);
+        ASSERT_GE(version.rgb.minor, (uint32_t)6);
+        ASSERT_GE(version.depth.major, (uint32_t)1);
+        ASSERT_GE(version.depth.minor, (uint32_t)6);
+        ASSERT_GE(version.audio.major, (uint32_t)1);
+        ASSERT_GE(version.audio.minor, (uint32_t)6);
+        ASSERT_EQ(version.firmware_build, K4A_FIRMWARE_BUILD_RELEASE);
+        ASSERT_EQ(version.firmware_signature, K4A_FIRMWARE_SIGNATURE_MSFT);
+    }
+
+    // should not throw exception
+    (void)kinect.is_sync_out_connected();
+    (void)kinect.is_sync_in_connected();
+
+    {
+        // should not throw exception
+        calibration cal = kinect.get_calibration(K4A_DEPTH_MODE_NFOV_2X2BINNED, K4A_COLOR_RESOLUTION_1440P);
+        calibration cal2 = cal;
+        ASSERT_EQ(cal.color_resolution, cal2.color_resolution);
+    }
+
+    {
+        std::vector<uint8_t> raw_cal = kinect.get_raw_calibration();
+        calibration cal = kinect.get_calibration(K4A_DEPTH_MODE_NFOV_2X2BINNED, K4A_COLOR_RESOLUTION_1440P);
+        calibration::get_from_raw(raw_cal.data(),
+                                  raw_cal.size(),
+                                  K4A_DEPTH_MODE_NFOV_2X2BINNED,
+                                  K4A_COLOR_RESOLUTION_1080P);
+        ASSERT_EQ(cal.color_resolution, K4A_COLOR_RESOLUTION_1080P);
+    }
+
+    {
+        k4a_color_control_mode_t mode;
+        int32_t value;
+        kinect.set_color_control(K4A_COLOR_CONTROL_EXPOSURE_TIME_ABSOLUTE, K4A_COLOR_CONTROL_MODE_AUTO, 0);
+        kinect.get_color_control(K4A_COLOR_CONTROL_EXPOSURE_TIME_ABSOLUTE, &mode, &value);
+        ASSERT_EQ(K4A_COLOR_CONTROL_MODE_AUTO, mode);
+    }
+
+    {
+        std::string sernum = kinect.get_serialnum();
+    }
+
+    {
+        k4a_imu_sample_t sample = { 0 };
+        capture cap1, cap2;
+        k4a_device_configuration_t config = K4A_DEVICE_CONFIG_INIT_DISABLE_ALL;
+        config.color_resolution = K4A_COLOR_RESOLUTION_1080P;
+        config.depth_mode = K4A_DEPTH_MODE_PASSIVE_IR;
+        config.synchronized_images_only = true;
+        kinect.start_cameras(&config);
+        kinect.start_imu();
+        ASSERT_TRUE(kinect.get_capture(&cap1));
+        ASSERT_TRUE(kinect.get_capture(&cap2));
+        ASSERT_TRUE(kinect.get_imu_sample(&sample));
+        ASSERT_TRUE(cap1 != cap2);
+        kinect.stop_cameras();
+        kinect.stop_imu();
+
+        ASSERT_NE(sample.acc_timestamp_usec, 0);
+        ASSERT_NE(sample.gyro_timestamp_usec, 0);
+
+        ASSERT_LT(0, cap1.get_temperature_c());
+        ASSERT_LT(0, cap2.get_temperature_c());
+        cap1.set_temperature_c(0.0f);
+        cap2.set_temperature_c(0.0f);
+
+        {
+            capture shallow_copy;
+            capture deep_copy;
+            shallow_copy = cap1;               // test = operator
+            ASSERT_TRUE(shallow_copy == cap1); // test == operator
+
+            deep_copy = std::move(cap1);       // test = (&&) operator
```

Comment:
>deep_copy = std::move(cap1) [](start = 12, length = 27)

This isn't a deep copy, it's a move - the object is never copied, it's just that the ownership of the handle changes

Reply:
you are right, I updated all the references to deep copy